### PR TITLE
Feature/low voltage

### PR DESCRIPTION
**Pull request by KhrystynaDubovyk**
Technical task : [#2233](https://github.com/smartdevicelink/sdl_core/issues/2233)
This PR is **ready** for review.
This pull request contains the implementation of the proposal : [0170-sdl-behavior-in-case-of-Low-Voltage](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0170-sdl-behavior-in-case-of-Low-Voltage.md)

### Risk
This PR makes **[no]** API changes.

### Testing Plan
Unit tests are provided
ATF tests are provided

### Summary

Implemented handling of LOW VOLTAGE, WAKE_UP,  IGNITION_OFF
signals based on real-time UNIX signals

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: docs/UnixSignals/LowVoltage/index.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#### SDL behavior in case of a `LOW_VOLTAGE` event:
+* SDL ignores all the requests from mobile applications without providing any kind of response.
+* SDL ignores all responses and messages from HMI except for `WAKE_UP` or `IGNITION_OFF` signals.
+* SDL stops audio/video streaming services.
+* All transports are unavailable for SDL.
+* SDL persists resumption related data stored before receiving a `LOW_VOLTAGE` signal.
+* SDL and the PoliciesManager must persist 'consumer data' (resumption-related + local PT) periodically and independently of the external events. 
+
+#### SDL resumes its regular work after receiving a "WAKE_UP" signal:
+* After receiving a `WAKE_UP` signal, all applications will be unregistered and the device disconnected.
```
Comment (AByzhynar) [inline docs/UnixSignals/LowVoltage/index.md]: @KhrystynaDubovyk This point is missed in diagram

Comment (KhrystynaDubovyk) [reply docs/UnixSignals/LowVoltage/index.md]: @AByzhynar, thank you
fixed with  https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/109/commits/7047ffd3a19230ba5c5af90345d89c7d1e0080d9, https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/109/commits/0264c5ebfc2e9ef429e6511afb0a12773ee62840

File: docs/UnixSignals/LowVoltage/index.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+UNIX signals provide ability to use signals from SIGRTMIN to SIGRTMAX for custom needs.  
+SDL uses this range for handling `LOW_VOLTAGE`, `WAKE_UP`, `IGNITION_OFF` notifications. 
+
+Offset for SIGRTMIN from this notifications is defined in the 'Main' section of the smartDeviceLink.ini file: 
+
+```
+[MAIN] 
+LowVoltageSignal = 1 ; Offset for from SIGRTMIN
+WakeUpSignal = 2 ; Offset for from SIGRTMIN
+IgnitionOffSignal = 2 ; Offset for from SIGRTMIN
```
Comment (yang1070) [inline docs/UnixSignals/LowVoltage/index.md]: typo, shall  be
IgnitionOffSignal = `3`

File: docs/UnixSignals/LowVoltage/index.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+3. Send `IGNITION_OFF` signal when SDL processes have to be terminated, applications have to be unregistered due SDL functionality suspension during a LOW VOLTAGE event
+!!!
+
+#### SDL behavior in case of a `LOW_VOLTAGE` event:
+* SDL ignores all the requests from mobile applications without providing any kind of response.
+* SDL ignores all responses and messages from HMI except for `WAKE_UP` or `IGNITION_OFF` signals.
+* SDL stops audio/video streaming services.
+* All transports are unavailable for SDL.
+* SDL persists resumption related data stored before receiving a `LOW_VOLTAGE` signal.
+* SDL and the PoliciesManager must persist 'consumer data' (resumption-related + local PT) periodically and independently of the external events.
```
Comment (yang1070) [inline docs/UnixSignals/LowVoltage/index.md]: the word `periodically` is confusing to me, does that mean PoliciesManager  can still do some writing to the disk periodically after receives a low voltage signal?

File: docs/UnixSignals/LowVoltage/index.md (hunk 2 above)
Comment (KhrystynaDubovyk) [reply docs/UnixSignals/LowVoltage/index.md]: @yang1070, thank you.
fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/109/commits/f596913186b6eb2dbbac6d774a8f851363133979

File: docs/UnixSignals/LowVoltage/index.md (hunk 3 above)
Comment (KhrystynaDubovyk) [reply docs/UnixSignals/LowVoltage/index.md]: @yang1070, thank you.
fixed f596913

**Comment by yang1070** [comment]
@theresalech Ford has reviewed and approved this PR. It is ready for Livio to review.

**Pull request merged**